addons/interaction/functions/fnc_takeLauncher.sqf

```diff
@@ -0,0 +1,46 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Timi007. Cplhardcore
+ * Pass launcher to another unit.
+ *
+ * Arguments:
+ * 0: Unit that passes the launcher <OBJECT>
+ * 1: Unit to pass the launcher to <OBJECT>
+ * 2: launcher classname <STRING>
+ * 3: Play passing animation <BOOL> (default: true)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_player, _target, "launch_NLAW_F"] call ace_interaction_fnc_passLauncher
+ *
+ * Public: No
+ */
+
+params ["_player", "_target", "_launcher", ["_animate", true, [true]]];
+TRACE_4("Pass launcher params",_player,_target,_launcher,_animate);
+
+if (_launcher isEqualTo "") exitWith {ERROR("No launcher specified.")};
+private _hasLauncher = secondaryWeapon _player;
+if (_hasLauncher == "") exitWith {ERROR("Cannot add launcher to target due to lack of inventory space.")};
+
+private _CfgWeapons = configFile >> "CfgWeapons" >> _launcher;
+private _attachments = _target weaponAccessories _launcher;
+private _launcherAmmo = (currentMagazine _unit) select {
+    _x in getArray (configFile >> "CfgWeapons" >> _launcherClassname >> "magazines")
+};
+_target removeWeapon _launcher;
+{
+        _unit removeMagazine _x;
+    } forEach _launcherAmmo;
+
+_player addWeapon _launcher;
```

Comment:
`addWeapon` automatically loads magazines that exist in a unit's inventory. The player could already have compatible ammo.

https://cbateam.github.io/CBA_A3/docs/files/common/fnc_addWeaponWithoutItems-sqf.html is a far better choice, handles removing all compatible magazines (including ones defines in magwells) and adding them back in the same containers they were in, with exact ammo counts.